core/src/main/scala/org/apache/spark/api/java/JavaPairRDD.scala

```diff
@@ -937,6 +937,28 @@ class JavaPairRDD[K, V](val rdd: RDD[(K, V)])
     fromRDD(new OrderedRDDFunctions[K, V, (K, V)](rdd).sortByKey(ascending, numPartitions))
   }
 
+  /**
+   * Return a RDD containing only the elements in the inclusive range `lower` to `upper`.
+   * If the RDD has been partitioned using a `RangePartitioner`, then this operation can be
+   * performed efficiently by only scanning the partitions that might containt matching elements.
+   * Otherwise, a standard `filter` is applied to all partitions.
+   */
+  def filterByRange(lower: K, upper: K): JavaPairRDD[K, V] = {
```

Comment:
You'll want the `@Since("3.1.0")` annotations on these methods

Reply:
Sure. I am also wondering whether it makes sense to backport this in 2.4 by the way?

Reply:
No, it's not a bug fix per se. I wouldn't put it in 3.0 even necessarily.

Reply:
Thank you for your first contribution, @wetneb !  +1 for @srowen 's answers.

Reply:
Just for curiosity, of the two since marks, Spark's scala @Since annotation and java @since tag, how to choose them? IMHO,  @since tag seems better here to let the version show up in the generated Java API documentation.  @dongjoon-hyun @srowen thanks.

Reply:
For the generated Java API doc,  you should put `@since` into the comment.
```
  /**
    * Return a RDD containing only the elements in the inclusive range `lower` to `upper`.
    * If the RDD has been partitioned using a `RangePartitioner`, then this operation can be
    * performed efficiently by only scanning the partitions that might containt matching elements.
    * Otherwise, a standard `filter` is applied to all partitions.
    * @since 3.1.0
    */
```

Reply:
Thanks, I have added that too.

Reply:
Yep that's right, my mistake, I was thinking of Scala